src/test/kotlin/pw/forst/tools/katlib/IterableExtensionsTest.kt

```diff
@@ -224,4 +224,24 @@ internal class IterableExtensionsTest {
         val result = listOf(10, 20).itemsToString("numbers", itemToString = itemToString )
         assertEquals("2 numbers: NUM10, NUM20", result)
     }
+
+    @Test
+    fun validateElements() {
+        val incrementingNumbers = listOf(2, 3, 4, 5, 6)
+        val differenceIsMaxOne = incrementingNumbers.validateElements { previousNumber, nextNumber ->
+            nextNumber - previousNumber == 1
+        }
+
+        assertEquals(true, differenceIsMaxOne)
+    }
+
+    @Test
+    fun flatMapIndexedNotNull() {
```

Comment:
could you please add a test which utilizes filtering `null`? thanks!

Reply:
Hi, in the recent commit I did include something that checked for nulls in the end result, but I'm not sure that is what you meant. If it wasn't, can you elaborate?

Reply:
yup, that one, thanks!